File: 프로그래머스/C++/LV1_모의고사.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    int arr1[5] = {1,2,3,4,5};
    int arr2[8] = {2,1,2,3,2,4,2,5};
    int arr3[10] = {3,3,1,1,2,2,4,4,5,5};
    int N = answers.size();
    int ans1 =0; int ans2 = 0; int ans3 = 0;
    for(int i =0 ; i<N; i++){
        if(arr1[i%5] == answers[i])  ans1++;
        if(arr2[i%8] == answers[i])  ans2++;
        if(arr3[i%10] == answers[i])  ans3++;
    }
    if(ans1 == ans2 && ans2 == ans3)    {
        answer.push_back(1);
        answer.push_back(2);
        answer.push_back(3);
    }else if(ans1 == ans2 && ans2 > ans3){
        answer.push_back(1);
        answer.push_back(2);
    }else if(ans1 == ans3 && ans3 > ans2){
        answer.push_back(1);
        answer.push_back(3);
    }else if(ans3 == ans2 && ans2 > ans1){
        answer.push_back(2);
        answer.push_back(3);
    }else if(ans1 > ans2 && ans1 > ans3){
        answer.push_back(1);
    }else if(ans2 > ans1 && ans2 > ans3){
        answer.push_back(2);
    }else{
        answer.push_back(3);
    }
    return answer;
}

File: 프로그래머스/JAVA/LV1_모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        
        int[] arr1 = {1,2,3,4,5};//size = 5
        int[] arr2 = {2,1,2,3,2,4,2,5};//size = 8
        int[] arr3 = {3,3,1,1,2,2,4,4,5,5};//size =10
        
        int score1 = 0;
        int score2 = 0;
        int score3 = 0;
        
        for(int i= 0; i<answers.length; i++){
            if(answers[i] == arr1[i%5]) score1++;
            if(answers[i] == arr2[i%8]) score2++;
            if(answers[i] == arr3[i%10]) score3++;
        }
        int ans_size;
        int[] ans = {1,2,3};
        if(score1 == score2 && score2 == score3)    ans_size = 3;
        else if(score1 == score2&&score2>score3){
            ans_size = 2;
            //ans[0] = 1; ans[1] = 2;
        }
        else if(score1 == score3 && score1>score2){
            ans_size = 2;
            ans[0] = 1; ans[1] = 3;
        }
        else if(score2 == score3 && score2>score1){
            ans_size = 2;
            ans[0] = 2; ans[1] = 3;
        }
        else{
            ans_size = 1;
            //if(score1>score2&&score1>score3)    ans[0] = 1;
            if(score2>score3&&score2>score1)    ans[0] = 2;
            if(score3>score2&&score3>score1)    ans[0] = 3;
        }
        
        int[] answer = new int[ans_size];
        for(int i = 0; i<ans_size; i++){
            answer[i] = ans[i];
        }
        return answer;
    }
}
